stablehlo/dialect/TypeInference.cpp

```diff
@@ -3175,10 +3094,10 @@ LogicalResult verifyAllReduceOp(std::optional<Location> location, Value operand,
 
   // all_reduce_c4
   if (useGlobalDeviceIds && channelId <= 0)
-    return emitOptionalError(
-        location,
-        "channel_id must be positive when useGlobalDeviceIds is set but got: ",
-        channelId);
+    return emitOptionalError(location,
+                             "channel_id must be positive when "
+                             "useGlobalDeviceIds is set but got: ",
+                             channelId);
 
   auto operandType = operand.getType().cast<ShapedType>();
   // all_reduce_c5
```

Comment:
Also in `verifyBitcastConvertOp`, let's simplify redundant cast:
```
  auto operandShapedType = operand.getType().cast<ShapedType>();
  auto targetShapedType = result.getType().cast<ShapedType>();
  ...
  auto operandType = operandShapedType.dyn_cast<RankedTensorType>();
  auto targetType = targetShapedType.dyn_cast<RankedTensorType>();
```

to:
```
auto operandType = operand.getType().cast<RankedTensorType>();
```

Reply:
Thanks for pointing that out, I'll do that in a follow-up PR.